pkg/k8s/watchers/node.go

```diff
@@ -127,25 +120,24 @@ func (k *K8sWatcher) GetK8sNode(_ context.Context, nodeName string) (*v1.Node, e
 // ciliumNodeUpdater implements the subscriber.Node interface and is used
 // to keep CiliumNode objects in sync with the node ones.
 type ciliumNodeUpdater struct {
-	k8sWatcher         *K8sWatcher
 	kvStoreNodeUpdater KVStoreNodeUpdater
 }
 
-func NewCiliumNodeUpdater(k8sWatcher *K8sWatcher, kvStoreNodeUpdater KVStoreNodeUpdater) *ciliumNodeUpdater {
+func NewCiliumNodeUpdater(kvStoreNodeUpdater KVStoreNodeUpdater) *ciliumNodeUpdater {
 	return &ciliumNodeUpdater{
-		k8sWatcher:         k8sWatcher,
 		kvStoreNodeUpdater: kvStoreNodeUpdater,
 	}
 }
 
 func (u *ciliumNodeUpdater) OnAddNode(newNode *v1.Node, swg *lock.StoppableWaitGroup) error {
-	u.updateCiliumNode(u.kvStoreNodeUpdater, newNode)
-
+	// We don't need to run OnAddNode because Cilium will fetch the state from
```

Comment:
Could we make the comment a bit more specific and say which component of Cilium will fetch state and populate KV store? I assume this is what's implemented in the operator here: https://github.com/cilium/cilium/blob/dbc538ef02ed34cbc7c401e2336c7ef02cb44409/operator/cmd/cilium_node.go#L52

Reply:
I've updated the comment with the two places where this is used.